osf/migrations/__init__.py

```diff
@@ -0,0 +1,133 @@
+        Q(codename='change_collectionprovider') |
+        Q(codename='delete_collectionprovider') |
+        Q(codename='change_providerassetfile') |
+        Q(codename='delete_providerassetfile') |
+        Q(codename='change_preprintrequest') |
+        Q(codename='delete_preprintrequest') |
+        Q(codename='change_registrationprovider') |
+        Q(codename='delete_registrationprovider')
+    )
+
+
+def get_admin_prereg_permissions():
+    from django.contrib.auth.models import Permission
+    return Permission.objects.filter(
+        Q(codename='view_prereg') |
+        Q(codename='administer_prereg')
+    )
+
+
+def update_admin_permissions(verbosity=0):
+    from django.contrib.auth.models import Group, Permission
+    should_log = verbosity > 0
+    # Create and add permissions for the read only group
+    group, created = Group.objects.get_or_create(name='read_only')
+    if created and should_log:
+        logger.info('read_only group created')
+    [group.permissions.add(perm) for perm in get_admin_read_permissions()]
+    group.save()
+    if should_log:
+        logger.info('View permissions added to read only admin group')
+
+    # Create  and add permissions for new OSF Admin group - can perform actions
+    admin_group, created = Group.objects.get_or_create(name='osf_admin')
+    if created and should_log:
+        logger.info('admin_user Group created')
+    [admin_group.permissions.add(perm) for perm in get_admin_read_permissions()]
+    [admin_group.permissions.add(perm) for perm in get_admin_write_permissions()]
+    group.save()
+    if should_log:
+        logger.info('Administrator permissions added to admin group')
+
+    # Create and add permissions for Prereg Admin group
+    prereg_group, created = Group.objects.get_or_create(name='prereg_admin')
+    if created and should_log:
+        logger.info('Prereg admin group created')
+    [prereg_group.permissions.add(perm) for perm in get_admin_prereg_permissions()]
+    prereg_group.save()
+    if should_log:
+        logger.info('Prereg read and administer permissions added to the prereg_admin group')
+
+    # Add a metrics_only Group and permissions
+    metrics_group, created = Group.objects.get_or_create(name='metrics_only')
+    if created and should_log:
+        logger.info('Metrics only group created')
+    metrics_permission = Permission.objects.get(codename='view_metrics')
+    metrics_group.permissions.add(metrics_permission)
+    metrics_group.save()
+
+    # Add a view_prereg Group and permissions
+    prereg_view_group, created = Group.objects.get_or_create(name='prereg_view')
+    if created and should_log:
+        logger.info('Prereg view group created')
+    prereg_view_permission = Permission.objects.get(codename='view_prereg')
+    prereg_view_group.permissions.add(prereg_view_permission)
+    prereg_view_group.save()
+
+
+def update_provider_auth_groups(verbosity=0):
+    # TODO: determine efficient way to only do this if perms change
```

Comment:
Maybe don't do this at all, but rely on `osf.management.commands.update_auth_groups` instead?

At the same time, with tables as small and relatively bounded as `AbstractProvider`, it shouldn't ever add more than a few seconds (timed locally at 3.52s).